### PR TITLE
[Reader IA] Integrate Top Bar Filter UI to Filter functionality

### LINKED ISSUE #19851: [Reader IA - Bug] Pull-to-refresh opens Discover Feed
While testing #19844 I found that the pull-to-refresh causes the feed to reset to "Discover".

Since we are working on a feature branch, I'm creating this issue to keep track of the bug so we fix it before making a final PR targeting `trunk`.

cc @RenanLukas 

### Expected Behavior
Pull-to-refresh refreshes the content of the current feed with the latest posts.

### Actual Behavior
Pull-to-refresh opens and refreshes the Discover feed.

### Steps to reproduce
On a build containing the changes from #19844, do the following:
1. Open and log into Jetpack
2. Go to `Reader`
3. Select a Feed that is not `Discover` in the top bar menu
4. Pull to refresh on the feed
5. **Verify** the `Discover` feed is opened and refreshed instead of the chosen feed

### LINKED ISSUE #19855: [Reader IA] Integrate Filter Chips with existing Filter functionality
```[tasklist]
### Tasks
- [x] Show correct filter chips for the feeds
- [x] Show the correct count in filter chips
- [x] Integrate filtering functionality with the Top Bar Filter component
- [x] Hide/Remove ViewPager inside Filter Bottom Sheet
```

### PR DESCRIPTION
**Pull request by thomashorta**
Fixes #19855 
Fixes #19851

> [!Caution]
> Should not be merged before #19844

This integrates the new Filter component with the existing Filter functionality, while also updating the Filter Sheet to look a bit closer to the new specs, but some things, like the Edit Button and the style update for site items will be done in future PRs.

As a side effect, this PR also fixes #19851 since the root cause of that bug also caused the feed to refresh when selecting a Filter, which was needed for this PR.

https://github.com/wordpress-mobile/WordPress-Android/assets/5091503/60f2c56d-6896-4e9c-8d1e-4fe4ab2ea653

-----

## To Test:

1. Install and log into Jetpack
2. Go to the Reader
3. Go to Feeds that are "filterable", such as "Subscriptions" and "Automattic" (for A12s)
4. **Verify** the appropriate filter chips appear, with the correct count inside them
5. Tap one of the filters
6. **Verify** the correct bottom sheet opens (for blogs or tags)
7. Select a filter item in the list
8. **Verify** the correct blog/tag title is shown when selected
9. Tap the filtered title (NOT the X button)
10. **Verify** the bottom sheet opens for the same type of item (blog/tag)
11. Cancel selection or select something else
12. Tap the X button in the Filter chip for the selected item
13. **Verify** clearing the filter works properly

-----

## Regression Notes

1. Potential unintended areas of impact

    - Fetching posts / Loading functionality inside the feeds

14. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual Testing

15. What automated tests I added (or what prevented me from doing so)

    - TODO

-----

## PR Submission Checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## UI Changes Testing Checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Review by develric — COMMENTED** [review]
Thanks for this PR @thomashorta 🙇 ! Overall it looks good. I'm reporting an issue I found in one of my test accounts though. While comparing the viewpager tabs and feeds content for before / after, I see a missing menu item for "Followed P2S" (that is a filterable with only Sites and no Tags elements, I think similarly to the Automattic item)



Not sure how common this scenario is, but I think that menu is related to P2 the product sites that I think I created in that account from [here](https://p2help.wordpress.com/getting-started/creating-a-new-workspace/) if it's useful to create the test scenario.

cc @RenanLukas fyi.

**Comment by thomashorta** [comment]
>  I'm reporting an issue I found in one of my test accounts though. While comparing the viewpager tabs and feeds content for before / after, I see a missing menu item for "Followed P2S" (that is a filterable with only Sites and no Tags elements, I think similarly to the Automattic item)

I don't think I understand what is the issue you're reporting here. In your comment, I only see a screenshot from the old UI (before) and there doesn't seem to be a report of what is the issue after the changes.

**Comment by develric** [comment]
Hey @thomashorta 👋 , it seems I missed to paste a picture of how it looks for me with the version on this PR, sorry if it was not clear from the description and here is a before/after 👍 

image

>While comparing the viewpager tabs and feeds content for before / after, I see a missing menu item for "Followed P2S" (that is a filterable with only Sites and no Tags elements, I think similarly to the Automattic item)

What I was trying to say here is (using the same user for the before/after):
- the "Followed P2s" item is missing in the after
- If it's helpful, note that the "Followed P2s" filtering had only Sites (and not Tags) and I'd expect this to be similar to the behavior of the Automattic menu (that I verified shows up correctly in the after as well)

About how to get the Followed P2s on your side, what follows is my best guess
>I think that menu is related to P2 the product sites that I think I created in that account from [here](https://p2help.wordpress.com/getting-started/creating-a-new-workspace/) if it's useful to create the test scenario.

but I can share my test account with you if it's quicker (just dm me in slack 👍 )

**Comment by thomashorta** [comment]
Ah, I understand it now, though I don't think it has anything to do with this PR since this is integrating the **filter** functionality (blogs and tags) with the real data from each field, and not related to the feeds and drop-down menu itself.

Regarding this:
> If it's helpful, note that the "Followed P2s" filtering had only Sites (and not Tags) and I'd expect this to be similar to the behavior of the Automattic menu (that I verified shows up correctly in the after as well)

The code in this PR already makes sure that we only show the "Blogs" pill for Automattic and P2 feeds, since I added the same logic used by the subscription management sheet to decide if the tag filter pill should be shown (https://github.com/wordpress-mobile/WordPress-Android/pull/19856/files#diff-2e098785089098d2a88d269ff91d4fb30d8f03c6c0fa67defdd892da6551e063R510-R511).

I think it makes sense to open a different issue for the "Followed P2" feed not showing on the drop-down menu, wdyt?

**Comment by thomashorta** [comment]
@develric I went ahead and created an issue for what you reported since it's not related to the code in this PR (https://github.com/wordpress-mobile/WordPress-Android/issues/19898).

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by RenanLukas** [comment]
Thanks for the PR, @thomashorta . I went through the testing steps and it's working as expected. I'm now moving to the code review.

I wanted to point out that [adding this call again](https://github.com/wordpress-mobile/WordPress-Android/pull/19856/commits/bc19aab80dc4005c97df3578e9bac51ac806510e) did not work for me to fix #19891:

https://github.com/wordpress-mobile/WordPress-Android/assets/14964993/fd19ef4f-c6ae-4c04-9350-9d3d665715bf

I've tried building locally and also with the APK provided by the `wpmobilebot`.

**Comment by RenanLukas** [comment]
As discussed in DM, there was another bug that was fixed that prevented the subscription blogs from being fetched (fixed [here](https://github.com/wordpress-mobile/WordPress-Android/pull/19856/commits/bc19aab80dc4005c97df3578e9bac51ac806510e)). #19891 will remain open so we can work on it when it's possible as a UX improvement to show something other than `"0 Blogs"` when there's a fresh install.

File: WordPress/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1623,9 +1623,8 @@
     <!-- reader -->
     <string name="reader">Reader</string>
     <string name="reader_filter_cta">Filter</string>
-    <string name="reader_filter_sites_title">Sites</string>
-    <string name="reader_filter_tags_title">Topics</string>
-    <string name="reader_filter_main_title">Following</string>
+    <string name="reader_filter_sites_title">Filter by blog</string>
```
Comment (RenanLukas) [inline WordPress/src/main/res/values/strings.xml]: I don't remember by heart: do we have to rename strings when the content changes to they're correctly picked up by GlotPress script?

File: WordPress/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1623,9 +1623,8 @@
...
+    <string name="reader_filter_tags_title">Filter by tag</string>
```
Comment (RenanLukas) [inline WordPress/src/main/res/values/strings.xml]: Same question about keeping the same string name when content changes.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/views/compose/filter/ReaderFilterChipGroup.kt [hunk 3]
```diff
@@ -63,31 +65,31 @@ fun ReaderFilterChipGroup(
     ) {
         val blogSelected = selectedItem?.type == ReaderFilterType.BLOG
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/views/compose/filter/ReaderFilterChipGroup.kt]: Nitpick: I know this wasn't implemented in this PR, but I missed it since I didn't review the PR that added it. Would it make sense to add a prefix to make it more obvious that it's a boolean and not the actual selected blog object?

e.g. `isBlogSelected`

File: WordPress/src/main/java/org/wordpress/android/ui/reader/views/compose/filter/ReaderFilterChipGroup.kt [hunk 4]
```diff
@@ -63,31 +65,31 @@ fun ReaderFilterChipGroup(
...
         val tagSelected = selectedItem?.type == ReaderFilterType.TAG
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/views/compose/filter/ReaderFilterChipGroup.kt]: Nitpick: same question about the prefix to make it more obvious it's a boolean.

e.g. `isTagSelected`

**Review by RenanLukas — COMMENTED** [review]
@thomashorta code LGTM 👍 

I just didn't approve yet to confirm if the strings with new content and same keys are a problem or not.

File: WordPress/src/main/res/values/strings.xml (hunk 1 above)
Comment (thomashorta) [reply WordPress/src/main/res/values/strings.xml]: If I remember correctly, the script (or maybe GlotPress) is smart enough to see the changes and mark the Strings as "untranslated" in the platform. cc @oguzkocer or @AliSoftware to confirm that though.

File: WordPress/src/main/java/org/wordpress/android/ui/reader/views/compose/filter/ReaderFilterChipGroup.kt (hunk 3 above)
Comment (thomashorta) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/views/compose/filter/ReaderFilterChipGroup.kt]: Done

File: WordPress/src/main/java/org/wordpress/android/ui/reader/views/compose/filter/ReaderFilterChipGroup.kt (hunk 4 above)
Comment (thomashorta) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/views/compose/filter/ReaderFilterChipGroup.kt]: Done

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/src/main/res/values/strings.xml (hunk 1 above)
Comment (AliSoftware) [reply WordPress/src/main/res/values/strings.xml]: We should always treat strings as immutable—once you created a key with a value, you should avoid changing its value later on, especially if the first value already made it to a previous release.

This is because otherwise once this change hits trunk it will be sent to GlotPress and the new copy retranslated… but if we then pull the translations from GlotPress into the `release/*` branch before submitting to Google then that change of the copy would end up making its way to the release branch… despite being a change tied to something in `trunk`.

In summary modifying the value of an existing key will cause a discrepancy in the `release/*` branch, between the codebase (which will expect the old copy  as that's what it was when code was frozen) and the translations (updated with the new copy when we'll pull the latest from GlotPress)

TL;DR: prefer using a brand new key.

Comment (RenanLukas) [reply WordPress/src/main/res/values/strings.xml]: Thanks, @AliSoftware 

I'll change the string keys with new content to brand new ones.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by RenanLukas — APPROVED** [review]
LGTM :shipit:

File: WordPress/src/main/res/values/strings.xml (hunk 1 above)
Comment (thomashorta) [reply WordPress/src/main/res/values/strings.xml]: Thanks @AliSoftware for the answer, that makes sense!
